common/app/renderers/DotcomRenderingService.scala

```diff
@@ -59,16 +59,21 @@ class DotcomRenderingService extends GuLogging with ResultWithPreconnectPreload
       ws: WSClient,
       payload: JsValue,
       endpoint: String,
+      fastlyRequestId: Option[String],
       timeout: Duration = Configuration.rendering.timeout,
   )(implicit request: RequestHeader): Future[WSResponse] = {
 
     val start = currentTimeMillis()
 
-    val resp = ws
+    val request = ws
       .url(endpoint)
       .withRequestTimeout(timeout)
       .addHttpHeaders("Content-Type" -> "application/json")
-      .post(payload)
+
+    val resp = fastlyRequestId match {
+      case Some(id) => request.addHttpHeaders("x-gu-xid" -> id).post(payload)
+      case None     => request.post(payload)
+    }
```

Comment:
Here, we won't add the `x-gu-xid` to the dcr request header if we don't have a value for it